src/CallbackContainer.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Dhii\Container;
+
+use Dhii\Container\Util\StringTranslatingTrait;
+use Psr\Container\ContainerInterface;
+use function call_user_func_array;
+
+/**
+ * A container implementation that decorates another container, invoking a callback for possible modifications to the
+ * decorated container's values.
+ *
+ * @since [*next-version*]
+ */
+class CallbackContainer implements ContainerInterface
```

Comment:
Another small point: when I see `CallbackContainer`, I think that there will be some callback that will return values based on a key. Not that the value of the inner container will be "filtered" via the callback. How about `FilterContainer` instead?

Reply:
Good point.

Reply:
Upon further consideration, I think the term "filter" could be confusing here. Filtering typically refers to the process of reducing the source set into a subset.

The proper term for changing every element of a source set into that of another set is "mapping". So what do you think about `MappingContainer`?

Reply:
Sounds good to me!